### PR TITLE
Remove support for Aliyun/Alibaba Cloud

### PR DESCRIPTION
**Pull request by johngmyers**
It's in the way of refactoring the `ClusterSpec` API.

### CODE REVIEW

File: docs/releases/1.24-NOTES.md [hunk 1]
```diff
@@ -24,8 +26,6 @@ This is a document to gather the release notes prior to the release.
 
 * The `node-role.kubernetes.io/master` and `kubernetes.io/role` labels are deprecated and might be removed from control plane nodes in future versions of kOps.
 
-* Due to lack of maintainers, the Aliyun/Alibaba Cloud support has been deprecated. The current implementation will be left as-is until the implementation needs updates or otherwise becomes incompatible. At that point, it will be removed. We very much welcome anyone willing to contribute to this cloud provider.
```
Comment (hakman) [inline docs/releases/1.24-NOTES.md:27]: This should be updated in 1.23 release notes to signal the change in 1.24.

File: docs/releases/1.24-NOTES.md [hunk 2]
```diff
@@ -12,6 +12,8 @@ This is a document to gather the release notes prior to the release.
 
 * Support for Kubernetes version 1.18 has been removed.
 
+* Support for Aliyun/Alibaba Cloud has been removed.
```
Comment (hakman) [inline docs/releases/1.24-NOTES.md:15]: There are some more reference in various docs to AliCloud/Aliyun/Alibaba. Should we remove those too?

Comment (johngmyers) [reply docs/releases/1.24-NOTES.md:15]: I didn't find any for "Aliyun" or "Alibaba" outside of release notes. I just removed references to "AliCloud".

File: docs/releases/1.24-NOTES.md (hunk 1 above)
Comment (johngmyers) [reply docs/releases/1.24-NOTES.md:27]: Updated.

File: docs/releases/1.24-NOTES.md (hunk 2 above)
Comment (hakman) [reply docs/releases/1.24-NOTES.md:15]: https://github.com/kubernetes/kops/blob/master/ROADMAP.md#upcoming-features ?

Comment (hakman) [reply docs/releases/1.24-NOTES.md:15]: Can be done later. That whole file need work.

**Review by hakman — APPROVED** [review]

**Pull request merged**